src/rgw/rgw_rest_pubsub.cc

```diff
@@ -497,6 +498,15 @@ void RGWPSCreateNotif_ObjStore_S3::execute(optional_yield y) {
   ps.emplace(static_cast<rgw::sal::RadosStore*>(store), s->owner.get_id().tenant);
   auto b = ps->get_bucket(bucket_info.bucket);
   ceph_assert(b);
+
+  // get all topics on a bucket
```

Comment:
move this inside the "if" in line 524. no need to do that unless we delete all notifications